xarray/core/indexing.py

```diff
@@ -543,6 +543,10 @@ def _updated_key(self, new_key):
         return _combine_indexers(self.key, self.shape, new_key)
 
     def __getitem__(self, indexer):
+        # load if the indexed array becomes a scalar
```

Comment:
Is this actually a good idea? I think it's a little surprising to load data immediately just because the result is a scalar.